src/documents/DataSourceAPI.md

```diff
@@ -0,0 +1,84 @@
+---
+title: DataSource API
+layout: default
+category: Infusion
+---
+
+The DataSource component provides the workflow for getting/setting (read/write) data from some data source. A data source may be an external database, cookie, in memory storage and etc. The specifics for accessing the data source must be provided in a concrete implementation. Additionally the payload of a request can be processed through an encoding step and other transformations during the get and set workflows.
+
+## Grades and Linkage
+
+There are two main grades `fluid.dataSource` and `fluid.dataSource.writable`. `fluid.dataSource` contains the base configuration and includes configuration for getting (reading) in from a data source. `fluid.dataSource.writable` adds in the configuration for setting (writing) to a data source. If you need to read and write to a data source, you'll need to add concrete implementations for both of these grades and make use of [grade linkage](IoCAPI.md#fluidmakegradelinkagelinkagename-inputnames-outputnames) to apply the concrete writable grade to the datasource configuration. In this way one would only need to add the `fluid.dataSource.writable` grade to gain the configuration from the concrete writable grade.
```

Comment:
Should clarify "If you need to read and write to a data source " .... to make clear that these are instructions for people who need to implement a *new* kind of DataSource. Most users should be able to get what they need by supplying configuration to the existing ones. In general the docs should make a clearer distinction between the needs of users configuring existing sources and those writing new ones and/or the concrete implementation for one.